sdk/messaging/azservicebus/example_receiver_test.go

```diff
@@ -86,3 +86,25 @@ func ExampleReceiver_ReceiveMessages() {
 		fmt.Printf("Received and completed the message\n")
 	}
 }
+
+func ExampleReceiver_DeadLetterMessage() {
+	// DeadLetterMessage settles a message by moving it to the dead letter queue.
+	ctx, cancel := context.WithTimeout(context.TODO(), 60*time.Second)
+	defer cancel()
+
+	message, err := receiver.ReceiveMessages(ctx, 1, nil)
```

Comment:
You get back a slice of messages here, not a single message. You'll need to check that the array is the length you want it to be and then dead letter only one message, not the entire slice.

Reply:
Something to this effect?

```go
	if len(messages) == 1 {
		err := receiver.DeadLetterMessage(context.TODO(), messages[0], deadLetterOptions)
		if err != nil {
			log.Fatal("%v", err)
		}
	}
```

Reply:
Updated the code to include a length check and deadletter a single message. :) Lmk if you'd like me to implement it differently.